### ISSUE TITLE
BinaryJedis.bitfield should return List<Long> instead of List<byte[]>

### ISSUE THREAD

**Issue by christophstrobl** [xetorthio/jedis#1413]
### Expected behavior

`BinaryJedis.bitfield` should return `List<Long>` instead of `List<byte[]>`. 
The same command issued via `PiplelineBase` or `JedisCommands` already returns the `Long` variant.

Writing a tests it seems that the return type declaration is wrong and that a `List<Long>` is provided already not matching the return types declaration. This leads to `java.lang.ClassCastException`.

#### Steps to reproduce: 

```java
List<byte[]> results = jedis.bitfield("mykey".getBytes(), new byte[][]{"SET".getBytes(), "i8".getBytes(), "0".getBytes(), "10".getBytes()});
for(byte[] result : results) {  // <-- fails
  System.out.println("result");
}
```

#### Jedis version: 
2.9.0

**Comment by marcosnils** [comment]
@christophstrobl thx for reporting. As this fails already I believe we can fix it ASAP in the next PATCH (2.9.1) version even though it breaks backwards compatibility (because it fails for everyone basically).

@HeartSaVioR WDYT?

**Comment by HeartSaVioR** [comment]
@marcosnils Yes agreed. Let's fix it and ship it with 2.9.1.
